PVTable Add two new plugins for poi and xmlbeans jar files.

charlesHenryPATARD opened the pull request:
Those two plugins need to be add in pvtable plugin dependencies.

berryma4 commented on plugins/org.csstudio.platform.libs.poi/.classpath:
```diff
@@ -0,0 +1,27 @@
+	<classpathentry exported="true" kind="lib" path="lib/poi-ooxml-3.13-20150929.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/poi-ooxml-schemas-3.13-20150929.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/poi-scratchpad-3.13-20150929.jar"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8">
```
@shroffk maybe we can _just_ add these to maven-osgi-bundles instead?

shroffk commented:
I am not sure if that would work.
I think there are significant issues associated with importing third party libraries through the maven osgi  bundles repo.

charlesHenryPATARD commented:
Hello,
@shroffk How to fix it ?

shroffk commented:
Henry, you could try to add these dependencies to the maven-osgi-bundles/repository/pom.xml
This will make them available as osgi-bundles which can be directly imported.

I would highly recommend that you test this first, I have faced significant difficulties with using the maven-osgi repo directly when libraries use java spi.

charlesHenryPATARD commented:
Hello,
I would like to be sure before create a new pull request. Just below you can see the list of files that I modified : 

- pom.xml of new plugins org.csstudio.platform.libs.poi and org.csstudio.platform.libs.xmlbeans. I do the same as pom.xml of org.csstudio.platform.libs.jersey
- I add org.csstudio.platform.libs.poi and org.csstudio.platform.libs.xmlbeans to thirdparty-plugins and modified the pom.xml
- I've modified the thirdparty-features/org.csstudio.thirdparty.all.feature

Can you confirm this before I create a pull request ? If I made something wrong can you explain me how to do this ? Thank you.

berryma4 commented:
I will add them to maven-osgi-bundles, although, I see a few of these at version 3.7.0 on orbit:
http://download.eclipse.org/tools/orbit/downloads/drops/R20151221205849/
It would be nice to use these, would the version 3.7.0 work for you?

berryma4 commented:
I've added apache poi 3.13 to the maven-osgi-bundles repo:
https://github.com/ControlSystemStudio/maven-osgi-bundles/commit/7c1c0a93d9df31c014684bcadbeedd01c20bcd4a
they will be available shortly from p2.

If this works for you, we will close this pull request.

charlesHenryPATARD commented:
Unfortunately org.apache.poi.xssf is missing from orbit. I've search all along the internet for up update site which contain this plugin. So, I really need to add org.csstudio.platform.libs.poi to thirdparty bundles.

berryma4 commented:
I believe org.apache.poi.xssf is in poi-ooxml, which I added to maven-osgi-bundles:
https://github.com/ControlSystemStudio/maven-osgi-bundles/commit/7c1c0a93d9df31c014684bcadbeedd01c20bcd4a

You shouldn't need this pull request, because we added the bundles to the 
http://download.controlsystemstudio.org/maven-osgi-bundles/4.3
repo

charlesHenryPATARD commented:
Ok,
Thank you. So, know everything is good for you ? Can we consider this job like done ?

berryma4 commented:
Does your PVTable pick up the libraries okay?  As @shroffk mentioned previously, osgi and spi don't always mix well (classloader issues).
so, PVTable importing the libraries from maven-osgi-bundles is working fine?
If so, i will close this.

kasemir commented:
Can this be closed?

berryma4 commented:
closed with the additions to maven-osgi
https://github.com/ControlSystemStudio/maven-osgi-bundles/commit/7b43bfd631f12e45f5365ccf774b571c7259d16d
